Carrito.js

```diff
@@ -0,0 +1,193 @@
+            this.eliminarDelCarrito(nombreProducto);
+            return;
+        }
+        this.guardarCarrito();
+        this.actualizarCarrito();
+    }
+
+    calcularTotal() {
+        const total = this.productos.reduce(
+            (acum, producto) => acum + producto.precio * producto.cantidad,
+            0
+        );
+        return this.formateador.format(total);
+    }
+
+
+    actualizarCarrito() {
+        const carritoContainer = document.querySelector("#carrito .cart-items");
+
+        carritoContainer.innerHTML = this.productos
+            .map(
+                (producto) => `
+                    <li class="carrito-item">
+                        <span class="carrito-nombre">${producto.nombre}</span>
+                        <span class="carrito-precio">$${this.formateador.format(producto.precio)}</span>
+                        <span class="carrito-cantidad">
+                            <button class="btn-decrementar" data-nombre="${producto.nombre}">-</button>
+                            ${producto.cantidad}
+                            <button class="btn-incrementar" data-nombre="${producto.nombre}">+</button>
+                        </span>
+                        <button class="btn-eliminar" data-nombre="${producto.nombre}">Eliminar</button>
+                    </li>
+                `
+            )
+            .join("");
+
+        document.querySelectorAll(".btn-decrementar").forEach((button) => {
+            button.addEventListener("click", () => this.decrementarCantidad(button.dataset.nombre));
+        });
+
+        document.querySelectorAll(".btn-incrementar").forEach((button) => {
+            button.addEventListener("click", () => this.incrementarCantidad(button.dataset.nombre));
+        });
+
+        document.querySelectorAll(".btn-eliminar").forEach((button) => {
+            button.addEventListener("click", () => this.eliminarDelCarrito(button.dataset.nombre));
+        });
+
+        const totalContainer = document.querySelector("#carrito-total span");
+        if (totalContainer) {
+            totalContainer.textContent = this.calcularTotal();
+        }
+    }
+
+    inicializarBotones() {
+        const botonesAgregar = document.querySelectorAll(".add-to-cart-button");
+
+        botonesAgregar.forEach((boton) => {
+            boton.addEventListener("click", () => {
+                const nombreProducto = boton.dataset.nombre;
+                const precioProducto = parseFloat(boton.dataset.precio);
+
+                if (!this.validarProducto(nombreProducto, precioProducto)) {
+                    console.error("Datos inválidos al agregar producto");
+                    return;
+                }
+
+                this.agregarAlCarrito(nombreProducto, precioProducto);
+            });
+        });
+    }
+
+    async cargarProductosAPI() {
+        try {
+            const respuesta = await fetch("https://fakestoreapi.com/products");
```

Comment:
Comprueba si la respuesta del fetch es exitosa con respuesta.ok antes de procesar el JSON. Esto maneja mejor los errores de red o de servidor.
```suggestion
       const respuesta = await fetch("https://fakestoreapi.com/products");
        if (!respuesta.ok) {
            throw new Error('Error al obtener datos de la API');
        }
        const productos = await respuesta.json();
```